### PR TITLE
CIRC-665: Record in house use data for each check-in.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
https://issues.folio.org/browse/CIRC-665: Record additional properties for each check-in

## Approach
Following additional properties have to be recorded during each check-in in order to record in-house use of an item:

- item status;
- Request queue size for item;
- Effective location for item.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/java/api/loans/CheckInByBarcodeTests.java [hunk 1]
```diff
@@ -872,12 +872,18 @@ private void verifyCheckInOperationRecorded(UUID itemId, UUID servicePoint) {
...
+
     recordedOperations.forEach(checkInOperation -> {
       assertThat(checkInOperation.getString("occurredDateTime"),
         withinSecondsBeforeNow(Seconds.seconds(2)));
       assertThat(checkInOperation.getString("itemId"), is(itemId.toString()));
       assertThat(checkInOperation.getString("servicePointId"), is(servicePoint.toString()));
       assertThat(checkInOperation.getString("performedByUserId"), is(getUserId()));
+      assertThat(checkInOperation.getString("itemStatusPriorToCheckIn"), is("Checked out"));
+      assertThat(checkInOperation.getString("itemLocationId"), is(itemEffectiveLocationId));
+      assertThat(checkInOperation.getInteger("requestQueueSize"), is(0));
```
Comment (marcjohnson-kint) [inline src/test/java/api/loans/CheckInByBarcodeTests.java]: Please can an example be added where there are open requests for this item

File: src/test/java/api/loans/CheckInByBarcodeTests.java [hunk 2]
```diff
@@ -872,12 +872,18 @@ private void verifyCheckInOperationRecorded(UUID itemId, UUID servicePoint) {
...
+    final String itemEffectiveLocationId = itemsClient.getById(itemId).getJson()
+      .getString("effectiveLocationId");
+
     recordedOperations.forEach(checkInOperation -> {
       assertThat(checkInOperation.getString("occurredDateTime"),
         withinSecondsBeforeNow(Seconds.seconds(2)));
       assertThat(checkInOperation.getString("itemId"), is(itemId.toString()));
       assertThat(checkInOperation.getString("servicePointId"), is(servicePoint.toString()));
       assertThat(checkInOperation.getString("performedByUserId"), is(getUserId()));
+      assertThat(checkInOperation.getString("itemStatusPriorToCheckIn"), is("Checked out"));
```
Comment (marcjohnson-kint) [inline src/test/java/api/loans/CheckInByBarcodeTests.java]: Please can an example be added for a status other than checked out

File: src/test/java/api/loans/InHouseUseCheckInTest.java [hunk 3]
```diff
@@ -120,4 +128,33 @@ public void isNotInHouseUseWhenCheckInServicePointIsNotServingHomeLocation() {
     assertThat(checkInResponse.getJson().containsKey("item"), is(true));
     assertThat(checkInResponse.getInHouseUse(), is(false));
   }
+
+  private void verifyLastCheckInRecordedAsInHouse(UUID itemId, UUID servicePoint) {
```
Comment (marcjohnson-kint) [inline src/test/java/api/loans/InHouseUseCheckInTest.java]: Given that it was decided that we would *not* record that a check in is considered to be in house use, I think this name is misleading.
```suggestion
  private void verifyLastCheckInWasRecorded(UUID itemId, UUID servicePoint) {
```

**Comment by marcjohnson-kint** [comment]
@bohdan-suprun Can the `do not merge` label be removed now that the storage changes have been merged?

File: src/test/java/api/loans/CheckInByBarcodeTests.java (hunk 2 above)
Comment (bohdan-suprun) [reply src/test/java/api/loans/CheckInByBarcodeTests.java]: There are tests in `src/test/java/api/loans/InHouseUseCheckInTest.java` that expects status to be `Available`. Should I added one more test to this file?

Comment (marcjohnson-kint) [reply src/test/java/api/loans/CheckInByBarcodeTests.java]: Apologies, given there are tests in another place that I missed, please disregard this comment

File: src/test/java/api/loans/InHouseUseCheckInTest.java [hunk 4]
```diff
@@ -120,4 +128,33 @@ public void isNotInHouseUseWhenCheckInServicePointIsNotServingHomeLocation() {
     assertThat(checkInResponse.getJson().containsKey("item"), is(true));
     assertThat(checkInResponse.getInHouseUse(), is(false));
   }
+
+  private void verifyLastCheckInWasRecorded(UUID itemId, UUID servicePoint) {
+    final CqlQuery query = queryFromTemplate("itemId=%s and itemStatusPriorToCheckIn=Available",
+      itemId);
+    final MultipleJsonRecords recordedOperations = checkInOperationClient.getMany(query);
+
+    assertTrue(recordedOperations.totalRecords() > 0);
```
Comment (marcjohnson-kint) [inline src/test/java/api/loans/InHouseUseCheckInTest.java]: Why was this approach chosen over using a `greaterThan` matcher?

Comment (bohdan-suprun) [reply src/test/java/api/loans/InHouseUseCheckInTest.java]: I've changed to use the method. Just was not aware of it.

**Pull request merged**